lib/ecto/repo/schema.ex

```diff
@@ -444,7 +444,14 @@ defmodule Ecto.Repo.Schema do
 
           # If there are no changes or all the changes were autogenerated but not forced, we skip
           {action, autogen} =
-            if original != %{} or (autogen != [] and force?),
+            if original != %{} ||
```

Comment:
I refactored this part to make the evaluation short-circuiting.
This was because I'm doing some array traversal inside the `forced_update?` function and I'm trying only to run it when necessary